src/Psalm/Internal/Analyzer/Statements/Expression/AssertionFinder.php

```diff
@@ -1559,47 +1561,106 @@ protected static function hasNotCountEqualityCheck(
      * @param PhpParser\Node\Expr\BinaryOp\Greater|PhpParser\Node\Expr\BinaryOp\GreaterOrEqual $conditional
      * @return false|int
      */
-    protected static function hasPositiveNumberCheck(
+    protected static function hasSuperiorNumberCheck(
         PhpParser\Node\Expr\BinaryOp $conditional,
-        ?int &$min_count
+        ?int &$literal_value_comparison,
+        bool &$isset_assert
     ) {
-        if ($conditional->right instanceof PhpParser\Node\Scalar\LNumber
-            && $conditional->right->value >= (
-                $conditional instanceof PhpParser\Node\Expr\BinaryOp\Greater
-                ? 0
-                : 1
-            )
-        ) {
-            $min_count = $conditional->right->value +
+        $right_assignment = false;
+        $value_right = null;
+        if ($conditional->right instanceof LNumber) {
```

Comment:
Note: if we have already analyzed `$right`/`$left` by the time we get here, it's probably better to use the already inferred type, so that it'll work with more complex expressions (e.g. `if ($a >= Limits::MAX + 1)`).

Reply:
Indeed, I'll check that. I wonder why it's not what's done in other cases...

Reply:
I did. However, the $source is not always a StatementsAnalyzer (on which we can retrieve type). That means I had to leave the rest of the conditionals for when it's something else, though I didn't check in what cases the type is different.

it may be the reason why Psalm didn't try with already inferred types before, though it seems to work fine so it may be extended to other assertions

Reply:
Note that if needed, we could go further. I only checked for Literal Ints but we should be able to use Positive Ints, or other ranges
for example if($range > $other_range) could be asserted to change $range to always be higher than $other_range (and the other way around too)